### PR TITLE
Composition

### PR DESCRIPTION
**Pull request by efoosag**
Implement Composition

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```
Comment (FrederickMih) [inline nameable.rb]: - [ ] Hi @efoosag, I suggest raising the error like this:

```.ruby 
         raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
```

File: nameable.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  end
+end
+
+class BaseDecorator < Nameable
```
Comment (FrederickMih) [inline nameable.rb:7]: - [ ] You'll need a getter for **nameable**, that's
```ruby
class BaseDecorator < Nameable
    attr_accessor :nameable
     
      ...


  
```
Kindly fix this issue :pray:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @efoosag , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

- Correct GitHub workflow :heavy_check_mark: 
- No Linter error :heavy_check_mark:  


## Required Changes ♻️

- [ ] In your README.md file, you'll provide instructions in the **Getting Started** section on how to set up your project locally. This is important for someone with fewer skills to set up your project in his local environment. Your Getting Started section should be replaced with the Build with section. Kindly follow the layout of this [Microverse README](https://github.com/microverseinc/readme-template) template. Also, fix the link in the License section because right now it's broken.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
### **Hi @efoosag,** 

### **YAY `Congratulations`! you are ready to move forward 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 
### **Note**

- **Please do make the PR very full of info because with it we will learn how you implemented your code.**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**